primo/data_parser/well_data.py

```diff
@@ -866,10 +874,41 @@ def _process_input_data(self):
 
     def _append_fed_dac_data(self):
         """Appends federal DAC data"""
-        census_year = self.config.census_year
-        # TODO:
-        # Append Tract ID/GEOID, population density, Total population, land area,
-        # federal DAC score to the DataFrame
+        if len(self.data) == 0:
+            # Nothing to do if no data
+            return
+
+        state_code = identify_state(self)
+        census_tracts = get_state_census_tracts(state_code, DAC_TRACT_YEAR)
+        gdf = get_data_as_geodataframe(self)
+
+        # Spatial join to identify tract id associated with every well
+        # Per 2010 data
+        gdf = gdf.sjoin(census_tracts, how="left", predicate="within")
+        self.data["Census Tract ID [2010]"] = pd.to_numeric(gdf["GEOID10"])
```

Comment:
Suggest replacing the line with: 
`self.add_new_column_ordered("geoid", "Census Tract ID [2010]", pd.to_numeric(gdf["GEOID10"]))`. This will register the column name in the column names object.
Could you also add a column containing the county code? This well help us do county-based clustering in the future.

Question: I'm hoping that there will not be an empty cells. But if there are a few, how do we handle them?

Reply:
There are no empty cells expected, but even if there are, I propose we still go forward with 0 as the DAC score but keeping the well under consideration

Reply:
Done. There is already a method utilized to extract the county code from the census tract, so that can be easily added when needed for clustering on county basis